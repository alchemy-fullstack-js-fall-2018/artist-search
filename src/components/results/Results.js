import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';

export default class Results extends Component {
  static propTypes = {
    currentPage: PropTypes.number.isRequired,
    updateCounts: PropTypes.func.isRequired,
    getResults: PropTypes.func.isRequired,
    ResultComponent: PropTypes.func.isRequired,
    query: PropTypes.object.isRequired
  };

  state = {
    results: []
  };

  updateResults = () => {
    const { currentPage, updateCounts, getResults, query } = this.props;
    getResults(query.name, currentPage)
      .then(({ totalPages, totalResults, results }) => {
        this.setState({ results });
        updateCounts(totalPages, totalResults);
      });
  };

  componentDidMount() {
    this.updateResults();
  }

  componentDidUpdate(prevProps) {
    if(prevProps.currentPage !== this.props.currentPage || prevProps.query !== this.props.query) {
      this.updateResults();
    }
  }

  render() {
    const { ResultComponent } = this.props;
    const { results } = this.state;

    const listItems = results.map(result => {
      return <ResultComponent key={result.id} result={result} />;
    });

    return (
      <Fragment>
        <div>
          <ul >
            {listItems}
          </ul>
        </div>
      </Fragment>
    );
  }
}
