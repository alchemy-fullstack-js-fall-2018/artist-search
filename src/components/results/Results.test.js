import React from 'react';
import { shallow } from 'enzyme';
import Results from './Results';

const updateCountFn = jest.fn();
const resultComponentFn = jest.fn();

const resultOfGetResults = {
  totalPages: 1,
  totalResults: 1,
  results: [{
    name: 'Joni Mitchell',
    description: 'American folksinger',
    type: 'Person',
    id: '12345'
  }]
};

const props = {
  currentPage: 1,
  updateCounts: updateCountFn,
  getResults: () => Promise.resolve(resultOfGetResults),
  ResultComponent: resultComponentFn,
  query: {
    name: 'Joni Mitchell'
  }
};

describe ('tests for Results component', () => {

  it('renders an Results component', () => {
    const wrapper = shallow(<Results {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });

});
