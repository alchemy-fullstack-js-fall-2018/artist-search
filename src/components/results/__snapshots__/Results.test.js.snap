// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for Results component renders an Results component 1`] = `
<Fragment>
  <div>
    <ul />
  </div>
</Fragment>
`;
